torch.compile: populate compiler_config

c00w opened the pull request:
Summary: This starts writing the compiler_config metadata into logger

Test Plan:
Modified existing test case to make sure this is not null.
(Also eyeballed what we're logging tomake sure it's reasonable

Reviewed By: masnesral

Differential Revision: D84014636




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

masnesral commented on torch/_dynamo/utils.py:
```diff
@@ -1553,6 +1554,27 @@ def clean_for_json(d: dict[str, Any]) -> dict[str, Any]:
+
+    if torch.compiler.config:
+        try:
+            compiler_config_copy = torch.compiler.config.get_config_copy()
```
lint complaint looks legit?

c00w committed:
torch.compile: populate compiler_config (#165581)

Summary:

This starts writing the compiler_config metadata into logger

Test Plan:
Modified existing test case to make sure this is not null.
(Also eyeballed what we're logging tomake sure it's reasonable

Reviewed By: masnesral

Differential Revision: D84014636
